src/js/actions/LibraryActions.js

```diff
@@ -71,6 +72,10 @@ export default {
             properties: ['openDirectory', 'multiSelections']
         }, (folders) => {
             if(folders !== undefined) {
+
+                folders.forEach((folder, index, array) => {
+                    array[index] = fs.realpathSync(folder);
```

Comment:
It is more like:
```js
Promise.map(folders, (folder) => {
  return new Promise((resolve, reject) => {
    fs.realpath(folder, (err, realpath) => {
      if (err) { return reject(err); }
      resolve(realpath);
    });
  })
}).then(resolvedFolders => {
  store.dispatch({
    type : AppConstants.APP_LIBRARY_ADD_FOLDERS,
    folders: resolvedFolders
  });
})
```

I'd also used [Bluebird's promisification features](http://bluebirdjs.com/docs/api/promisification.html) to avoid doing 
```js
return new Promise((resolve, reject) => { ... });
```